src/main/java/com/basho/riak/client/api/commands/search/Search.java

```diff
@@ -153,7 +153,7 @@ else if (option.getKey() == Option.DEFAULT_OPERATION)
             builder.withStart(start);
         }
 
-        if (rows > 0)
+        if (rows >= 0)
```

Comment:
I think I see now.  You changed the value to be initialized to -1 so now you have to set it to zero explicitly.